HIP-0007.md

```diff
@@ -0,0 +1,214 @@
+4. DANE is slow: decentralized systems are never convenient and there are difficult
+trade-offs for adoption. Simple technology like caching can help improve this lag.
+
+## Motivation
+
+We propose a secure, simplified subset of the DANE protocol that can secure the
+entire blockchain-naming ecosystem. Its properties are chosen to provide state-of-the-art
+cryptographic security and limited flexibility to reduce the technical effort required
+by web browsers to implement it.
+
+## Goals
+
+1. Do not alter or affect in any way the resolution, authentication, or fetching
+of ICANN-rooted domain names.
+
+2. When browsing to a decentralized domain name, indicate to the user whether or not
+the connection is secure (just like for legacy domains). A connection MUST
+be indicated as secure ONLY if ALL requirements of the specification are satisfied.
+
+3. Reduce the complexity (and increase the cryptographic security) of implementing
+DANE in browsers by limiting available options.
+
+## Specification
+
+### DNS Resolution
+
+This protocol does not require that browsers resolve domain names internally, although
+some networks like Handshake and Namecoin have embedable light clients: [hnsd] [ncds].
+It is expected that users will already have their system configured to resolve names
+by a trusted resolver (e.g. validating proof-of-work) that is either maintained by themselves
+or optionally, by a browser vendor for public use. Such a resolver MUST verify DNSSEC
+and ONLY set the DNS `ad` bit when DNSSEC is validated according to this specification.
+If the `ad` bit is not set, the protocol has failed and the user should be alerted that
+the connection is insecure. DNSSEC validity MUST be checked by the browser for BOTH
+the location record (`A`/`AAAA`) and the certificate matching record (`TLSA`).
+No other DNS records need to be examined by the browser internally.
+
+**We do not require the browser itself to resolve domain names or verify DNSSEC, we
+only require it to check that an `ad` bit has been set on two queries (location and
+certificate match).**
+
+### Root KSK
+
+The security of "legacy" DNS, DNSSEC and DANE comes down to single key: ICANN's
+[KSK2017]. Since all decentralized naming systems replace this security model
+with their own protocol, this key is not part of our system. However since we
+want to approach DANE with as much legacy compatibility as possible, some kind of
+root KSK must be trusted by the resolver that verifies DNSSEC.
+
+For example, the Handshake full node reference implementation [hsd] has a built-in
+root authoritative name server which signs all requests with a "public private key"
+(after verifying proof-of-work). This private key is hard-coded into the software
+and because it is public, the user must have a secure connection with a trusted HNS
+resolver.
+
+That connection can be secured with:
+
+- DNS-over-HTTPS: In this context, "HTTPS" relies entirely on legacy DNS and certificate
```

Comment:
Is this a constraint or just a starting presumption? If we already had good DANE support an alternative way, what would keep us from using that to bootstrap DoH? I'm concerned that "not recommended" phrasing implies that it would offer no advantage, but there are important reasons for it being adopted in terms of user traffic privacy etc. so I'd like to do everything we can to make it viable. 

Aside from simply depending on the below options for bootstrapping the DoH connection, I could also imagine there are other applicable ways of distributing/pinning a trusted cert fingerprint like just coding it into the browser like google does for themselves, web of trust peer distribution methods, etc. -- maybe a little more of a config chore but I'd prefer to give people the option.